addons/sale_loyalty/models/sale_order.py

```diff
@@ -337,10 +337,7 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
         # discount should never surpass the order's current total amount
         max_discount = min(self.amount_total, max_discount)
         if reward.discount_mode == 'per_point':
-            points = self._get_real_points_for_coupon(coupon)
-            if not reward.program_id.is_payment_program:
-                # Rewards cannot be partially offered to customers
-                points = points // reward.required_points * reward.required_points
+            points = self._get_real_points_for_coupon(coupon) if reward.program_id.is_payment_program else reward.required_points
```

Comment:
What? Why?? :thinking: 
As long as "clear all promo points" isn't True, the program could be applied multiple time, why force the use of only one?

Reply:
Also, could you try to wrap your lines at 100 chars? =] Thank you

Reply:
> What? Why?? 🤔 As long as "clear all promo points" isn't True, the program could be applied multiple time, why force the use of only one?

From what I v understood while communicating with the po (dala) if it's a gift card or e-wallet, all points should be used until we reach 0 loyalty point or we discount the whole product.

Reply:
Yes, I was mostly reacting to the else statement here, as you would only take the number of points of the `required_points`, and not any multiple of it anymore

Reply:
I thought that was the bug. The fact that it automatically apply multiple times. I force it to get used once to be able to re-select in the interface if we need it once again.

Reply:
It's clearly intended, even the test that you are modifying hint to that. So, I won't allow modifying that kind of comportment in stable, as people are using it and now expecting it. Now for the master, maybe something could be done to have a feature easier to use.

cc: @dalaOdoo your opinion might be required here

Reply:
Hello @dalaOdoo . Maybe I didn't fully understand when we had a conversation about the problem. So the ticket doesn't represent a bug?

Reply:
Sorry, I miss understood the bug. I updated the commit to fix just the partial reward. What do you think about this version of fix @Demesmaeker